File: growing-shop-auth-before/src/main/java/com/example/growingshopauth/config/error/GlobalExceptionHandler.java
package com.example.growingshopauth.config.error;

import com.example.growingshopauth.config.error.exception.InvalidJwtTokenException;
import com.example.growingshopauth.config.error.exception.NotAllowPathException;
import com.example.growingshopauth.config.error.exception.NotFoundUserException;
import com.example.growingshopcommon.config.error.ErrorResponse;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@RestControllerAdvice
public class GlobalExceptionHandler {

    @ExceptionHandler(NotFoundUserException.class)
    @ResponseStatus(HttpStatus.UNAUTHORIZED)
    public ErrorResponse notFoundUserException(NotFoundUserException exception) {
        return new ErrorResponse(exception.getMessage());
    }

    @ExceptionHandler(InvalidJwtTokenException.class)
    @ResponseStatus(HttpStatus.UNAUTHORIZED)
    public ErrorResponse invalidJwtTokenException(InvalidJwtTokenException exception) {
        return new ErrorResponse(exception.getMessage());
    }

    @ExceptionHandler(NotAllowPathException.class)
    @ResponseStatus(HttpStatus.UNAUTHORIZED)
    public ErrorResponse notAllowPathException(NotAllowPathException exception) {
        return new ErrorResponse(exception.getMessage());
    }
}

File: growing-shop-auth/src/main/kotlin/com/example/growingshopauth/config/error/GlobalExceptionHandler.kt
package com.example.growingshopauth.config.error

import com.example.growingshopauth.config.error.exception.InvalidJwtTokenException
import com.example.growingshopauth.config.error.exception.NotAllowPathException
import com.example.growingshopauth.config.error.exception.NotFoundUserException
import com.example.growingshopcommon.config.error.ErrorResponse
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class GlobalExceptionHandler {

    @ExceptionHandler(NotFoundUserException::class)
    @ResponseStatus(HttpStatus.UNAUTHORIZED)
    fun notFoundUserException(exception: NotFoundUserException): ErrorResponse {
        return ErrorResponse(exception.message)
    }

    @ExceptionHandler(InvalidJwtTokenException::class)
    @ResponseStatus(HttpStatus.UNAUTHORIZED)
    fun invalidJwtTokenException(exception: InvalidJwtTokenException): ErrorResponse {
        return ErrorResponse(exception.message)
    }

    @ExceptionHandler(NotAllowPathException::class)
    @ResponseStatus(HttpStatus.UNAUTHORIZED)
    fun notAllowPathException(exception: NotAllowPathException): ErrorResponse {
        return ErrorResponse(exception.message)
    }

    @ExceptionHandler(IllegalAccessException::class)
    @ResponseStatus(HttpStatus.UNAUTHORIZED)
    fun illegalAccessException(exception: IllegalAccessException): ErrorResponse {
        return ErrorResponse(exception.message)
    }
}
